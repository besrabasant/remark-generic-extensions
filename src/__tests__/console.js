import test from "ava"
import sinon from "sinon"

import { transformToHtml } from "./helpers/transform"

let originalConsoleLog

test.beforeEach(setup)
test.afterEach(cleanup)

function setup() {
  originalConsoleLog = console.log
  console.log = sinon.spy()
}

function cleanup() {
  console.log = originalConsoleLog
}

test(
  "should log debug messages to console when debug is enabled",
  t => {
    transformToHtml(
      "!icon",
      {
        debug: true
      }
    )
    t.true(console.log.callCount === 10)
    for (let index = 0; index < console.log.callCount; index++) {
      const args = console.log.getCall(index).args
      t.snapshot(args.toString())
    }
  }
)

test(
  "should not log debug messages to console by default",
  t => {
    transformToHtml("!icon")
    t.true(console.log.callCount === 0)
  }
)
