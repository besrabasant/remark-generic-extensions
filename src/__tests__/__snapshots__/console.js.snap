// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should log debug messages to console when debug is enabled 1`] = `"Element type:,icon"`;

exports[`should log debug messages to console when debug is enabled 2`] = `"Full match:,!icon"`;

exports[`should log debug messages to console when debug is enabled 3`] = `"Content:,undefined"`;

exports[`should log debug messages to console when debug is enabled 4`] = `"The content property is not mapped to any html attribute."`;

exports[`should log debug messages to console when debug is enabled 5`] = `"Argument:,undefined"`;

exports[`should log debug messages to console when debug is enabled 6`] = `"The argument property is not mapped to any html attribute."`;

exports[`should log debug messages to console when debug is enabled 7`] = `"Id:,undefined"`;

exports[`should log debug messages to console when debug is enabled 8`] = `"Class names:,undefined"`;

exports[`should log debug messages to console when debug is enabled 9`] = `"Properties:,undefined"`;

exports[`should log debug messages to console when debug is enabled 10`] = `
"
"
`;
